controllers/volumereplicationgroup_controller.go

```diff
@@ -1277,6 +1291,42 @@ func (ObjectStorePVUploader) UploadPV(v interface{}, s3ProfileName string,
 	return nil
 }
 
+func (v *VRGInstance) deletePVFromS3Stores(pvc *corev1.PersistentVolumeClaim, log logr.Logger) error {
+	log.Info("Delete PV from s3 stores", "PV", pvc.Spec.VolumeName, "s3Profiles", v.instance.Spec.S3ProfileList)
+
+	for _, s3ProfileName := range v.instance.Spec.S3ProfileList {
+		if err := v.reconciler.PVDeleter.DeletePV(v, s3ProfileName, pvc); err != nil {
+			return fmt.Errorf("error deleting PV %s using profile %s, err %w", pvc.Name, s3ProfileName, err)
+		}
+	}
+
+	return nil
+}
+
+type ObjectStorePVDeleter struct{}
+
+func (ObjectStorePVDeleter) DeletePV(v interface{}, s3ProfileName string,
+	pvc *corev1.PersistentVolumeClaim) (err error) {
+	vrgName := v.(*VRGInstance).instance.Name
+	s3Bucket := constructBucketName(v.(*VRGInstance).instance.Namespace, vrgName)
+
+	v.(*VRGInstance).log.Info("Deleting PersistentVolume metadata from object store", "Profile Name", s3ProfileName)
+
+	objectStore, err := v.(*VRGInstance).reconciler.ObjStoreGetter.objectStore(
+		v.(*VRGInstance).ctx, v.(*VRGInstance).reconciler, s3ProfileName, vrgName)
+	if err != nil {
+		return fmt.Errorf("failed to get client for s3Profile %s, err %w",
+			s3ProfileName, err)
+	}
+
+	// Delete PV to object store
+	if err := objectStore.deleteObject(s3Bucket, pvc.Spec.VolumeName); err != nil {
```

Comment:
Is there a way to detect "not found" errors when deleting?

So that if delete reconcile runs again and attempts to delete the PV we can treat that as a non-error.

We would modify this to let go of the PVC last as per #220 at which point we may run through half processed PVCs/PVs.

Reply:
We'll find a suitable error and return success when the PV metadata is not found.

Reply:
Currently we list and then delete, so should be safe to avoid the not found (IOW we can track the not found and fix it later, but if we are deleting the bucket we can work with the bucket not found error).